File: 3n+1/Main.java
import java.util.Scanner;
import java.lang.Math;

public class Main{

    public static void main(String[] args) {
    
        Scanner scanner = new Scanner(System.in);
        while(scanner.hasNext()) {
            int i = scanner.nextInt();
            int j = scanner.nextInt();
            int maiorNum = Math.max(i,j);
            int menorNum = Math.min(i,j);
            int maiorCiclo = 1;
            for(int n = menorNum; n <= maiorNum; n++) {
                int tamanhoCiclo = calcAlgo(n);
                if (tamanhoCiclo > maiorCiclo) maiorCiclo = tamanhoCiclo;
            }
            System.out.println(i + " " + j + " " + maiorCiclo);
        }
    }

    private static int calcAlgo(int n) {
        int tamanhoCiclo = 1;
        while (n != 1) {
            n = (n % 2 != 0) ? 3 * n + 1 : n / 2;
            tamanhoCiclo++;
        }
        return tamanhoCiclo;
    }
}

File: 3n+1/main.py
def calcAlgo(n):
    cycle_length = 1
    while n != 1:
        n = 3*n + 1 if n%2 != 0 else n/2
        cycle_length += 1
    return cycle_length
raw = raw_input().split(' ')
if len(raw) != 2 or int(raw[0]) <= 0 or int(raw[1]) >= 1000000: exit()
i,j = [int(n) for n in raw]
maiorCiclo = 1
menorNum,maiorNum = sorted([i,j])
for n in range(menorNum,maiorNum+1):
    ciclo_len = calcAlgo(n)
    if ciclo_len > maiorCiclo: maiorCiclo = ciclo_len
print('{} {} {}'.format(i,j,maiorCiclo))